### PR TITLE
8345074: java.net.InterfaceAddress constructor could be made private

### PR DESCRIPTION
**Pull request by eirbjo**
Please review this PR which suggests to make the constructor of `java.net.InterfaceAddress` private.

This constructor is only accessed by native code via the `NetworkInterface` class. Making it private would express the intent of this class as being non-subclassable.

The comment of the constructor is updated to reflect the new access modifier as well as to make it clear that the constructor is not to be invoked from Java code and that instances are obtained from `NetworkInterface` via native code. 

Marking this class final is handled separately in JDK-8344943 via a CSR.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8345074](https://bugs.openjdk.org/browse/JDK-8345074): java.net.InterfaceAddress constructor could be made private (**Enhancement** - P4)

### Reviewers
 * [Jaikiran Pai](https://openjdk.org/census#jpai) (@jaikiran - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22467/head:pull/22467` \
`$ git checkout pull/22467`

Update a local copy of the PR: \
`$ git checkout pull/22467` \
`$ git pull https://git.openjdk.org/jdk.git pull/22467/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22467`

View PR using the GUI difftool: \
`$ git pr show -t 22467`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22467.diff">https://git.openjdk.org/jdk/pull/22467.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22467#issuecomment-2509136519)
</details>

### CODE REVIEW

**Comment by eirbjo** [comment]
Verification: Ran tests such as `NetworkPrefixLength` to verify that native code can access the now-private constructor.

Stack trace in debugger:

```
        at java.base/java.net.InterfaceAddress.<init>(InterfaceAddress.java:49)
	at java.base/java.net.NetworkInterface.getAll(Native Method)
	at java.base/java.net.NetworkInterface.getNetworkInterfaces(NetworkInterface.java:354)
```

File: src/java.base/share/classes/java/net/InterfaceAddress.java [hunk 1]
```diff
@@ -42,10 +42,10 @@ public class InterfaceAddress {
     private short        maskLength = 0;
 
     /*
-     * Package private constructor. Can't be built directly, instances are
-     * obtained through the NetworkInterface class.
+     * Private constructor. Not to be constructed by Java code, instances
+     * are obtained from native code through the NetworkInterface class.
```
Comment (liach) [inline src/java.base/share/classes/java/net/InterfaceAddress.java:46]: We should word like "this constructor is called via JNI in NetworkInterface.c" so we know what care we need to take when we update this constructor.

Comment (eirbjo) [reply src/java.base/share/classes/java/net/InterfaceAddress.java]: > We should word like "this constructor is called via JNI in NetworkInterface.c" so we know what care we need to take when we update this constructor.

Thanks! Yes, simpler to just say where the constructor is called from. It folllows from this that instances are not meant to be created from Java code.

Pointing out that it's private adds little value as its now obvious from the access modifier.

**Review by jaikiran — APPROVED** [review]
This looks OK to me. I'll run our CI tests against this change, so please wait to see how that goes before integrating.

**Comment by jaikiran** [comment]
A tier1, tier2, tier3 run in our CI with this change went fine without any related failures.

**Comment by eirbjo** [comment]
Thanks a lot for reviews and CI testing, much appreciated!

I'll go ahead and integrate this now.

/integrate

**Pull request closed without merging**